### ISSUE TITLE
OpenSSL/libssl update

### ISSUE THREAD

**Issue by Machion** [zeld/psa-update#6]
Hi, I recently run your program and got a "error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory". 
Somehow my OS (Manjaro) dropped support for OpenSSL 1 in favour for OpenSSL 3, because some month ago it worked without problems. Luckily v1 is still in the repos, so I just had to install it again. 
But maybe you should consider updateing to OpenSSL 3 too, because OpenSSL 1 is only supported till 11th September 2023.

**Comment by zeld** [comment]
Hi, thanks for the heads up.

OpenSSL requirement comes from [Reqwest crate](https://github.com/seanmonstar/reqwest) which I expect to be already compliant with OpenSSL 3 although this is not stated in their README.

So my assumption is that this is rather a build issue and that binaries built with github runners were compiled against OpenSSL 1.1. With upgrade of ubuntu runners to version 22.04 that happened end of 2022 I would expect next binaries to be built against OpenSSL 3. I will track this for next release.

Workaround could be to build from source.

**Comment by zeld** [comment]
I can confirm that building on my machine works fine with OpenSSL 3. I'll check for next release that binaries produced by Giltlab are also built against Open SSL 3.

**Comment by zeld** [comment]
I released new binaries in version 1.0.1 that should now work with OpenSSL 3

```console
$ ldd psa-update 
	linux-vdso.so.1 (0x00007ffef2fd7000)
	libssl.so.3 => /lib/x86_64-linux-gnu/libssl.so.3 (0x00007f515b15b000)
	libcrypto.so.3 => /lib/x86_64-linux-gnu/libcrypto.so.3 (0x00007f515ac00000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f515b866000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f515b072000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f515a800000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f515b89d000)

```
